libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -810,8 +829,15 @@ bool ContractCompiler::visit(EmitStatement const& _emit)
 
 bool ContractCompiler::visit(VariableDeclarationStatement const& _variableDeclarationStatement)
 {
-	StackHeightChecker checker(m_context);
 	CompilerContext::LocationSetter locationSetter(m_context, _variableDeclarationStatement);
+
+	// Local variable slots are reserved when their declaration is visited,
+	// and freed in the end of their scope.
+	for (auto _decl: _variableDeclarationStatement.declarations())
+		if (_decl)
+			addScopedVariable(*_decl);
```

Comment:
This currently always zero-initializes variables, even if they are assigned to in the same statement, we might optimize that in a future PR.

Reply:
Yes, I'll create an issue an assign to myself to keep track of it.